formats/html.ts

```diff
@@ -21,24 +73,340 @@ export class HtmlImporter extends FormatImporter {
 			return;
 		}
 
-		let results: ImportResult = {
-			total: 0,
+		const results = await Promise.all(files
+			.map(file => this.processFile(folder, file)));
```

Comment:
> * Memory limits. If we work in parallel, this could lead to out-of-memory cases, especially for concurrently downloading multiple attachments. This is a much bigger problem on mobile devices, if/when we want to support them in the future.

For memory limits, I believe this can be mitigated by limiting the number of processing files and attachments that can run at once.

> * One-time use. Because the use case of the importer isn't a day-to-day thing, I think most people will be ok to wait a few more seconds for their import. I'd want to make sure we don't do anything obviously inefficient, but I think maximum parallelization isn't something we need to achieve.

Can't argue against that. (Though I imagine a few wanting to use multiple tools, making frequent use of it...)

> For this specific importer there's also a potential issue of rate-limiting since we are downloading some of these files from the internet.

Same as memory limits. In fact, I have already limited the max number of concurrent http connections to 5: https://github.com/obsidianmd/obsidian-importer/blob/453d8371ddff574e0c7b85f6f2fff66bcdc2a7d7/formats/html.ts#L12-L13

---

Here's a patch that adds a "parallel" toggle to the settings and logs the time elapsed for comparing concurrent and sequential processing:
- [parallel-toggle.patch](https://github.com/obsidianmd/obsidian-importer/files/12248430/parallel-toggle.patch) (base: 453d8371ddff574e0c7b85f6f2fff66bcdc2a7d7)

Testing indicates that concurrent processing speeds up reading local files and http requests, and doesn't do much when reading and writing files in the vault (probably because `Vault` operations are forced to be serial). So it depends on the amount of local attachments and external attachments. Also, it looks like Turndown, the tool for turning HTML into Markdown, takes quite some time (0.1~1 s per file), so I think the best way would be to sequentially process Markdown files, and download attachments concurrently.